src/wasm.rs

```diff
@@ -249,3 +249,24 @@ pub async fn read_parquet_stream(
     });
     Ok(wasm_streams::ReadableStream::from_stream(stream).into_raw())
 }
+
+#[wasm_bindgen(js_name = "transformParquetStream")]
+#[cfg(all(feature = "writer", feature = "async"))]
+pub fn transform_parquet_stream(
```

Comment:
Is `transform` the usual name for writing to a stream? E.g. why is this not `write_parquet_stream`?

Reply:
> Is `transform` the usual name for writing to a stream? E.g. why is this not `write_parquet_stream`?

I've had a bit of time to think this one through, and it largely boils down to convention/platform semantics, basically the use-case is split between three scenarios:

#### `ReadableStream<RecordBatch> -> ReadableStream<ArrayBuffer>`

The Streams spec preferences the `pipeThrough` approach, which involves any of:
1. `inputStream.pipeThrough(new TransformStream({transform(), start(), flush()}))` - best for trivial transforms that occur in JS (e.g. uppercasing text)
2. `inputStream.pipeThrough(new TransformSubClassStream(...params))` - for more involved internals that merit defining a class, or browser-provided native subclasses (e.g. `CompressionStream`, `TextDecoderStream`).
3. `inputStream.pipeThrough({writable, readable})` - any object with a writable and readable property (both of the correct type). Useful when you can't explicitly subclass from TransformStream.

It turns out that the following design for a TransformStream works flawlessly (the second internal TransformStream seems to have near-zero overhead):

```mermaid
flowchart TB
 subgraph B1["TransformStream"]
    direction TB
        f1["readable"]
        i1["writable"]
  end
 subgraph B2["TransformStream"]
    direction TB
        f2["readable"]
        i2["writable"]
  end
 subgraph TOP["ParquetEncoderStream"]
    direction LR
        B1
        nm["Stream&lt;RecordBatch&gt;"]
        ny[["Loop:<br>Encode Parquet"]]
        B2
  end
    i1 --> f1
    i2 --> f2
    A["Input Stream:<br>ReadableStream&lt;RecordBatch&gt;"] -- pipeThrough --> TOP
    B1 -- dyn_into --> nm
    nm --> ny
    ny --> B2
    TOP --> B["Output Stream:<br>ReadableStream&lt;ArrayBuffer&gt;"]
    style i1 fill:#FFD600,color:#000000
    style f2 fill:#FFD600,color:#000000
```

(yellow blocks being wasm-bindgen members of the struct, accessible from JS)

#### `Table -> ReadableStream<ArrayBuffer>`

Quite similar to a callable equivalent of `Response::body` - you'd typically provide `Table::stream()`, `Table::parquetStream()` for method-based apis, or `toParquetStream(Table)` for functional apis.

#### Rust: `Stream<RecordBatch> -> ReadableStream<ArrayBuffer>`

For downstream rust dependencies that expose something along the lines of `runPipeline(targetUrl)`, or `foo_state_machine.toParquetStream()` to JS, and need a means of getting the final rust stream of record batches back out to a readable stream of bytes. Used by the JS-facing transform stream internally.

Reply:
So I think with all that said, we'd have 3 exports:

- JS, `ParquetEncoderStream`, with usage `inputStream.pipeThrough(new ParquetEncoderStream(writerProperties))`
- JS, `writeParquetStream`, with usage: `writeParquetStream(inputTable)`
- RS, `to_parquet_stream`, with usage: `to_parquet_stream(rs_stream_of_record_batches)`

The sticking point on the last two exports is really that you can't do function overloads, and reasonably accurate distinguished function names would be too verbose (e.g. writeTableToParquetStream/writeTableParquetStream,  writeStreamToParquetStream :grimacing: )